### PR TITLE
Small Refactorings

### PR DESCRIPTION
**Pull request by calebnelton**
Was not paying attention in WebEx meeting ... this was more fun.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs [hunk 1]
```diff
@@ -395,10 +395,7 @@ protected virtual PropertyModel CreatePropertyModel(PropertyInfo propertyInfo)
                 // public void Foo() { }
                 //
                 // Is one action.
-                if (!(routeProvider is IActionHttpMethodProvider))
-                {
-                    createActionForSilentRouteProviders = false;
-                }
+                createActionForSilentRouteProviders &= routeProvider is IActionHttpMethodProvider;
```
Comment (Eilon) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs]: Not personally a fan of the readability or debuggability of this (it was easy to set a breakpoint in the `if()` before).

**Comment by Eilon** [comment]
@calebnelton thanks for the PR. Can you squash everything into one commit?

I should note that we generally prefer avoiding such changes because they don't offer a significant improvement of the code, and it takes a fair bit of time from many people to work on reviewing, merging, and verifying the PR. Feel free to discuss with us through issues any additional changes you might have in mind.

File: src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/BindingSource.cs [hunk 2]
```diff
@@ -183,7 +183,7 @@ public virtual bool CanAcceptDataFrom(BindingSource bindingSource)
         /// <inheritdoc />
         public bool Equals(BindingSource other)
         {
-            return other == null ? false : string.Equals(other.Id, Id, StringComparison.Ordinal);
+            return other != null && string.Equals(other.Id, Id, StringComparison.Ordinal);
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/BindingSource.cs]: `public bool Equals(BindingSource other) => string.Equals(other?.Id, Id, StringComparison.Ordinal);`

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs [hunk 3]
```diff
@@ -592,8 +589,7 @@ protected virtual bool IsAction(TypeInfo typeInfo, MethodInfo methodInfo)
             var routeTemplateProvider =
                 attributes
                 .OfType<IRouteTemplateProvider>()
-                .Where(a => !IsSilentRouteAttribute(a))
-                .SingleOrDefault();
+                .SingleOrDefault(a => !IsSilentRouteAttribute(a));
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs]: This should probably be `FirstOrDefault`.cc  @rynowak

Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs]: That would hide a bug. It's intended that this method only be called when one or zero attributes is a non-silent route provided. TLDR all of this code is complicated on purpose.

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs]: Should we throw a nicer exception in that case? `SingleOrDefault`'s message isn't particularly intuitive.

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs]: Regardless, this change is fine as is.

**Comment by pranavkm** [comment]
:watch:

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs (hunk 1 above)
Comment (calebnelton) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs]: I'll remove it when I rebase and squash into a single commit.

**Comment by calebnelton** [comment]
@Eilon Squashed commits

Code comment from @pranavkm incorporated also

**Comment by pranavkm** [comment]
Great. I'll merge this in soon

**Comment by pranavkm** [comment]
Merged in https://github.com/aspnet/Mvc/commit/5b0b0b95a7d6d40c4506abb105186042934a9b4b

**Comment by pranavkm** [comment]
Thanks for the PR!

**Pull request closed without merging**